projects/plugins/jetpack/modules/widget-visibility/editor/index.jsx

```diff
@@ -0,0 +1,284 @@
+/**
+ * WordPress dependencies
+ */
+import { Fragment, useEffect, useCallback, useMemo } from '@wordpress/element';
+import { Button, SelectControl, ToggleControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { InspectorAdvancedControls } from '@wordpress/block-editor'; // eslint-disable-line import/no-unresolved
+import { Icon, close } from '@wordpress/icons';
+
+/* global widget_conditions_data */
+/* eslint-disable react/react-in-jsx-scope */
+
+const blockHasVisibilitySettings = name => {
+	// When I put extra attributes on these blocks, they
+	// refuse to render with a message "Error loading block: Invalid parameter(s): attributes"
+	// However, most blocks don't do this. Why is this?
+	const disallowed = new Set( [ 'core/archives', 'core/latest-comments', 'core/latest-posts' ] );
+	return ! disallowed.has( name );
+};
+
+const VisibilityRule = props => {
+	const { rule, onDelete, setMajor, setMinor } = props;
+
+	const majorOptions = [
+		{ label: __( '-- Select --', 'jetpack' ), value: '' },
+		{ label: __( 'Category', 'jetpack' ), value: 'category' },
+		{ label: __( 'Author', 'jetpack' ), value: 'author' },
+		{ label: __( 'User', 'jetpack' ), value: 'loggedin' }, // TURN OFF FOR WPCOM
+		{ label: __( 'Role', 'jetpack' ), value: 'role' }, // TURN OFF FOR WPCOM
+		{ label: __( 'Tag', 'jetpack' ), value: 'tag' },
+		{ label: __( 'Date', 'jetpack' ), value: 'date' },
+		{ label: __( 'Page', 'jetpack' ), value: 'page' },
+		{ label: __( 'Taxonomy', 'jetpack' ), value: 'taxonomy' }, // ONLY IF A NON-DEFAULT TAXON IS FOUND
+	];
+
+	let minorOptions = [];
+	if ( rule.major in widget_conditions_data ) {
+		minorOptions = widget_conditions_data[ rule.major ].map( ( [ x, y ] ) => ( {
+			value: x,
+			label: y,
+		} ) );
+		// TODO: *Page* has nested arrays
+		// We don't handle these yet
+		// [
+		//     [ "front", "Front page" ],
+		//     [ "posts", "Posts page" ],
+		//     [ "archive", "Archive page" ],
+		//     [ "404", "404 error page" ],
+		//     [ "search", "Search results" ],
+		//     [
+		//         "Post type:",
+		//         [
+		//             [ "post_type-post", "Post" ],
+		//             [ "post_type-page", "Page" ],
+		//             [ "post_type-attachment", "Media" ]
+		//         ]
+		//     ],
+		// ]
+	}
+
+	return (
+		<div className="widget-vis__rule widget-vis__flex">
+			<div className="widget-vis__rule-col-1">
+				<div className="widget-vis__flex">
+					<div className="widget-vis__delete-rule">
+						<Button onClick={ onDelete }>
+							<Icon icon={ close } size={ 14 } />
+						</Button>
```

Comment:
I don't think this will be accessible, a screen reader won't be able to tell what this button does. Adding a label prop will address that.